python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -1213,31 +1211,52 @@ def _reorder_maps(
             Right gather map
         right_policy
             Nullify policy for right map
+        maintain_order
+            Which DataFrame row order to preserve
 
         Returns
         -------
         list of reordered left and right gather maps.
-
-        Notes
-        -----
-        For a left join, the polars result preserves the order of the
-        left keys, and is stable wrt the right keys. For all other
-        joins, there is no order obligation.
         """
         dt = plc.interop.to_arrow(plc.types.SIZE_TYPE)
         init = plc.interop.from_arrow(pa.scalar(0, type=dt))
         step = plc.interop.from_arrow(pa.scalar(1, type=dt))
-        left_order = plc.copying.gather(
-            plc.Table([plc.filling.sequence(left_rows, init, step)]), lg, left_policy
-        )
-        right_order = plc.copying.gather(
-            plc.Table([plc.filling.sequence(right_rows, init, step)]), rg, right_policy
-        )
+        if maintain_order in {"none", "left_right", "right_left"}:
```

Comment:
*issue/question*: If we have no obligation `maintain_order == "none"` I think we should not be doing any work, what is happening here?

Reply:
Yes you're correct, I'll also need to update the other tests in the suite since [polars defaults to "none" ](https://docs.pola.rs/api/python/stable/reference/dataframe/api/polars.DataFrame.join.html#polars.DataFrame.join). So I'll add `maintain_order="left"` to ensure those tests are reproducible.